include/dsn/c/api_layer1.h

```diff
@@ -578,7 +578,7 @@ typedef union dsn_global_partition_id
 
 inline uint64_t dsn_gpid_to_hash(dsn_gpid gpid)
 {
-    return (((uint64_t)gpid.u.app_id) << 32) + gpid.u.partition_index;
+    return static_cast<uint64_t>(gpid.u.app_id ^ gpid.u.partition_index);
```

Comment:
oh, the std one is unfortunately required for those unordered collections such as unordered_map. So it will not affect our partition and thread localization. Anymore, it is best to make them consistent.